### DISCUSSION TITLE
Unexpected behaviour when using $watch

### DISCUSSION THREAD

**Discussion by felixfrey** [alpinejs/alpine#2480] [7. Bugs (archived)]
Hi,

I am building a thing where there is a parent component and some child components. The following, strange behaviour occurs, when the code in the `$watch` closure inside the children is executed: the expressions inside x-show attributes elements inside the children are not evaluated correctly. Two elements with `x-show` expressions, one of the form `a === b`, the other one of the negated form `a !== b`  are both evaluated to be true after the function inside `$watch`was executed!

For a better understanding, I build a JSFiddle: [https://jsfiddle.net/7ymaco9g/26/](https://jsfiddle.net/7ymaco9g/26/)
To reproduce the described behaviour, read the code, click the **toggle parentState** button and watch the chaos unfold, both 
**states NOT equal** AND **states equal** are shown at the same time!

Am I using `$watch` in an unintended way, is there an obvious flaw in my example or is this a bug?

Best regards,
Felix

**Comment by SimoTod** [comment]
I believe the reason is that the show phase happens after 1 tick (so it doesn't conflict with click.outside) while the hide phase is not delayed. your code triggers a show a hide in the same frame so the show always wins even when the condition is not verified anymore.
